src/devices/Ws2812b/Ws2812b.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Spi;
+using System.Drawing;
+
+namespace Iot.Device.Ws2812b
+{
+    /// <summary>
+    /// Driver for WS2812B. Intelligentcontrol LED integrated light source
+    /// </summary>
+    public class Ws2812b : IntegratedLed
+    {
+        /// <summary>
+        /// Spi clock frequency for WS2812B
+        /// </summary>
+        public const int SpiClockFrequency = 3000000;
+
+        private SpiDevice _spiDevice;
+        private readonly byte[] _buffer;
+
+        private const byte Bit0 = 0b0100; // 333ns low(dummy), 333ns high(T0H), 667ns low(T0L)
+        private const byte Bit1 = 0b1100; // 667ns high(T1H), 667ns low(T1L)
+        private const byte ResetDelay = 105; // 280us low
+
+        private const byte ColorSize = 4;
+        private const byte PixelSize = ColorSize*3;
+
+        /// <summary>
+        /// Initializes a new instance of the WS2812B device.
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="length">Number of LEDs</param>
+        public Ws2812b(SpiDevice spiDevice, int length)
+        {
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+            _pixels = new Color[length];
```

Comment:
the constructor of the abstract class should take the length instead and allocate this buffer. If you're gonna make array public make sure to only make getter available publicly (i.e. `public Color[] Pixels { get; private set; }`)